website/pages/blog/aws-encryption-and-multi-account-access.md

```diff
@@ -0,0 +1,270 @@
+            },
+            "Action": [
+                "kms:Describe*",
+                "kms:Get*",
+                "kms:List*"
+            ],
+            "Resource": "*"
+        }
+    ]
+}
+```
+
+Let’s take a deeper look at this key policy and what this statement does.  Since they’re both Allow statements, these 2 statements function as a logical or statement for access.  This Key Policy has 2 main statements:
+
+- “Allow access through S3 for all principals in the account that are authorized to use S3”
+
+```json
+{
+            "Sid": "Allow access through S3 for all principals in the account that are authorized to use S3",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "*"
+            },
+            "Action": [
+                "kms:Encrypt",
+                "kms:Decrypt",
+                "kms:ReEncrypt*",
+                "kms:GenerateDataKey*",
+                "kms:DescribeKey"
+            ],
+            "Resource": "*",
+            "Condition": {
+                "StringEquals": {
+                    "kms:CallerAccount": "123412341234",
+                    "kms:ViaService": "s3.us-east-1.amazonaws.com"
+                }
+            }
+        },
+```
+
+There are 2 main conditions on this statement that limit the access to this key, despite the “AWS”: “*” Principal block.
+
+- “kms:CallerAccount”: “123412341234”
+- “kms:ViaService”: “s3.us-east-1.amazonaws.com”
+
+By combining the kms:CallerAccount condition with a Principal element that specifies all AWS identities, this statement specifies all identities in an AWS account, in this case the account 123412341234.  The kms:ViaService then limits the usage of an AWS KMS Key to requests from a specified AWS service, in this case “s3.us-east-1.amazonaws.com.”  Thus, this key can only be used by all identities in the 123412341234 account with requests from “s3.us-east-1.amazonaws.com.”
+
+- “Allow direct access to key metadata to the account”
+
+```json
+{
+            "Sid": "Allow direct access to key metadata to the account",
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "arn:aws:iam::123412341234:root"
+            },
+            "Action": [
+                "kms:Describe*",
+                "kms:Get*",
+                "kms:List*"
+            ],
+            "Resource": "*"
+        }
+```
+
+With the Principal in this statement being “AWS”: “arn:aws:iam::123412341234:root”, the account principal, the statement doesn’t give any IAM users or roles permissions to use the KMS key.  Instead, this allows for delegation of permissions.  Thus, if an IAM role or user has the above permissions to Describe, Get, and List the key, the IAM role or user can Describe,Get, and List the key.  This is because key policies require explicit permissions on the key policy (different from other resource policies) to grant permissions.  This allows for the key metedata access to be managed via IAM policies.
```

Comment:
```suggestion
With the Principal in this statement being “AWS”: “arn:aws:iam::123412341234:root”, the account principal, the statement doesn’t give any IAM users or roles permissions to use the KMS key.  Instead, this allows for delegation of permissions.  Thus, if an IAM role or user has the above permissions to Describe, Get, and List the key, the IAM role or user can Describe,Get, and List the key.  This is because key policies require explicit permissions on the key policy (different from other resource policies) to grant permissions.  This allows for the key metadata access to be managed via IAM policies.
```